### ISSUE TITLE
`nvidiaLegacy390` doesn't build on latest 19.03

### ISSUE THREAD

**Issue by vaibhavsagar** [NixOS/nixpkgs#62012]
## Issue description

`nvidiaLegacy390` doesn't build with `linuxPackages_4_19` or `linuxPackages_latest`.

### Steps to reproduce

```bash
[vaibhavsagar@nixos:~/repos/nixpkgs]$ git rev-parse HEAD
70a494a48627a96354e37644a124ec9cb3f31e68

[vaibhavsagar@nixos:~/repos/nixpkgs]$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A linuxPackages_4_19.nvidia_x11_legacy390
<...>
applying patch /nix/store/xf2x05y9q5scdck8aqynj6kb8nc5phcg-drm_mode_connector.patch
patching file kernel/nvidia-drm/nvidia-drm-connector.c
Hunk #1 FAILED at 226.
1 out of 1 hunk FAILED -- saving rejects to file kernel/nvidia-drm/nvidia-drm-connector.c.rej
patching file kernel/nvidia-drm/nvidia-drm-encoder.c
Hunk #1 FAILED at 216.
1 out of 1 hunk FAILED -- saving rejects to file kernel/nvidia-drm/nvidia-drm-encoder.c.rej
builder for '/nix/store/xgfxhh50gvfhv956f2x21ix9y2n2lhxi-nvidia-x11-390.116-4.19.45.drv' failed with exit code 1
error: build of '/nix/store/xgfxhh50gvfhv956f2x21ix9y2n2lhxi-nvidia-x11-390.116-4.19.45.drv' failed

[vaibhavsagar@nixos:~/repos/nixpkgs]$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A linuxPackages_latest.nvidia_x11_legacy390
<...>
make[3]: *** [/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source/scripts/Makefile.build:276: /build/kernel/nvidia/nv-instance.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source/scripts/Makefile.build:276: /build/kernel/nvidia/nv-gpu-numa.o] Error 1
make[3]: *** [/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source/scripts/Makefile.build:276: /build/kernel/nvidia/nv-frontend.o] Error 1
make[3]: *** [/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source/scripts/Makefile.build:276: /build/kernel/nvidia/nv.o] Error 1
make[2]: *** [/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source/Makefile:1571: _module_/build/kernel] Error 2
make[2]: Leaving directory '/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/build'
make[1]: *** [Makefile:169: sub-make] Error 2
make[1]: Leaving directory '/nix/store/af0vnrririr8k2kf2zvqisfsckqbgrsz-linux-5.1.4-dev/lib/modules/5.1.4/source'
make: *** [Makefile:79: modules] Error 2
builder for '/nix/store/pmyvqvn9857pq8d63afs185rddw5crfb-nvidia-x11-390.116-5.1.4.drv' failed with exit code 2
error: build of '/nix/store/pmyvqvn9857pq8d63afs185rddw5crfb-nvidia-x11-390.116-5.1.4.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by flokli** [comment]
In the case of nvidia, we need to add patches for more recent kernels.

That's what other Distros do too: https://git.archlinux.org/svntogit/packages.git/tree/trunk?h=packages/nvidia-390xx

In the case of debian, they seem to reside in https://salsa.debian.org/nvidia-team/nvidia-graphics-drivers, but behind a loginwall.

Can you have a look at https://github.com/NixOS/nixpkgs/pull/62064?

**Comment by vaibhavsagar** [comment]
Fixed by #62064.
